perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -1911,9 +1913,15 @@ sub redirect_or_confirm_creation : Private {
             $c->log->info($report->user->id . ' is an inspector - redirecting straight to report page for ' . $report->id);
             $c->res->redirect( $report->url );
         } else {
-            $c->log->info($report->user->id . ' was logged in, showing confirmation page for ' . $report->id);
+            $c->log->info($report->user->id . ' was logged in, redirecting to confirmation page for ' . $report->id);
             $c->stash->{created_report} = 'loggedin';
             $c->stash->{template} = 'tokens/confirm_problem.html';
+            unless ($no_redirect) {
+                my $redirect = $c->uri_for_action( '/report/confirmation', [ $report->id ] );
```

Comment:
Could factor this + copy below into another method on Problem to return a report's URI (would need to pass in $c this one though, I guess, for `uri_for_action`).